### PR TITLE
Added the Dolphin

### PR DESCRIPTION
**Pull request by Mailaender**
http://cnc.wikia.com/wiki/Dolphin_(Red_Alert_2)

### CODE REVIEW

File: rules/allied-naval.yaml [hunk 1]
```diff
@@ -129,3 +129,85 @@ aegis:
...
+	Cloak:
+		CloakTypes: Underwater
+		InitialDelay: 50
+		CloakDelay: 75
+		CloakSound: vnavupa.wav
+		UncloakSound: vnavupa.wav
+		WhileCloakedUpgrades: underwater
+		UncloakOnAttack: false
+	Targetable:
+		TargetTypes: Ground, Water, Repair
```
Comment (abcdefg30) [inline rules/allied-naval.yaml]: Why `Repair` here (and below)?

File: rules/allied-naval.yaml [hunk 2]
```diff
@@ -129,3 +129,85 @@ aegis:
...
+	DetectCloaked:
+		CloakTypes: Underwater
+		Range: 4c0
+	HiddenUnderFog:
+	ActorLostNotification:
+	Guard:
+		Voice: Attack
+	Guardable:
+	DeathSounds@normal:
+		DeathTypes: ExplosionDeath, BulletDeath, FlameDeath, ElectroDeath, RadiationDeath
```
Comment (abcdefg30) [inline rules/allied-naval.yaml]: You shouldn't need to define all that. If `DeathTypes` is empty a sound will always play: https://github.com/OpenRA/OpenRA/pull/10156

File: rules/allied-naval.yaml (hunk 1 above)
Comment (Mailaender) [inline rules/allied-naval.yaml]: Copy pasta. Removed it.

File: rules/allied-naval.yaml (hunk 2 above)
Comment (Mailaender) [inline rules/allied-naval.yaml]: Good to know. I had a long fight with the linter about this.

File: weapons.yaml [hunk 3]
```diff
@@ -783,6 +783,42 @@ SubTorpedo:
...
+		AffectsParent: false
+		ValidStances: Neutral, Enemy
+		Versus:
+			Heavy: 80
+			Concrete: 60
+		DamageTypes: Prone50Percent, TriggerProne, ExplosionDeath
+	Warhead@2Dam: SpreadDamage
+		Range: 0, 32
+		Falloff: 50, 50
+		Damage: 10
```
Comment (GraionDilach) [inline weapons.yaml]: The other warhead has 10 damage. Decide.

**Comment by Mailaender** [comment]
Optimized the weapon a bit.

**Comment by abcdefg30** [comment]
Ok, just some (minor) things:
- The weapon animation doesn't look right, but I guess we can't cut the trail at the moment?
- It uncloaks to fire, that should not happen
- What about using `NavalUnitLost` as `ActorLostNotification`?

Looks good to me otherwise.

**Comment by Mailaender** [comment]
Yeah, I think this needs some adjustments to the laser rendering to make it cut the trail and gain more transparency. The game ignoring `UncloakOnAttack: false` is probably an upstream engine bug. Nothing I can do about these here. 

Added `NavalUnitLost`.

**Comment by Mailaender** [comment]
Took https://github.com/OpenRA/OpenRA/pull/9808 into account to fix the uncloak problem and also slowed down the Soviet naval yard animations.

**Comment by abcdefg30** [comment]
Looks good to me. Thanks. :+1:

**Pull request merged**